lib/WorseReflection/Core/Reflection/Collection/ReflectionClassLikeCollection.php

```diff
@@ -55,6 +57,11 @@ public static function fromNode(ServiceLocator $serviceLocator, TextDocument $so
 
             if ($child instanceof ClassDeclaration) {
                 $items[(string) $child->getNamespacedName()] = new ReflectionClass($serviceLocator, $source, $child, $visited);
+                continue;
+            }
+
+            if ($child instanceof ObjectCreationExpression && !($child->classTypeDesignator instanceof Node)) {
```

Comment:
this may cause unexpected issues. "reflect classes in" reflects locatable symbols and also it _should_ not need to deeply traverse the AST.

we could consider removing this and instead adding support to `$reflector->reflect{Node,AnnonymousClass,Object}($node)` for example.

Reply:
The problem is this approach won't find all classes anyways. Because we actually have to traverse into class definitions as well. Otherwise we would miss anonymous classes inside other classes. But I don't have an idea how to do improve performance here.

Reply:
yes, they need probably to be interpreted inline as types as with the above comment. doing this will mean that "implement contracts" will not automatically work, but at the same time it's only working because of the cache effect in this PR

Reply:
I just saw that `reflectClassesIn` should probably better be named `reflectClassLikeIn` 'cause it also returns interfaces enums etc. which are completely irrelevant for the "Implement contracts". But that's not a problem of this PR.

However I get what you're saying, I'm not sure if its based on cache in this situation because the source code builder reflects the same structure again and should also build the same list of `ClassLike` nodes. On the other hand I'm quite familiar with the concept of inline types and how I would go about and implement this. Do we already have an example for this?